### DISCUSSION TITLE
ESPHome Plant waterer discussion

### DISCUSSION THREAD

**Discussion by rasclatt-dot-com** [rasclatt-dot-com/ESP32-Plant-Waterer-for-ESPHome#2] [General]
## 👋 Welcome!
  Using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Being supportive - there's no need for putdowns just because you may be more knowledgeable - sharing is how everyone learns and how DIY home automation communities thrive!

**Comment by stompy88** [comment]
hi is it possible to use one atom lite and a few pumps and sensors ?

**Reply by nborson** [reply]
I am using one atom lite for two pumps and sensors. As far as I can tell, that's the maximum because (besides the Grove connector) only GPIO33 has the ADC capability required for a moisture sensor.

**Comment by drkunze** [comment]
The example .yaml includes:
# Enable logging
logger:

Sorry for newbie question, but what goes into "logger" ?

Thanks

**Reply by nborson** [reply]
Hello @drkunze. If I understand your question right, nothing needs to go into it; just the one line is enough to enable logging with default settings.
`logger:`
There are various _optional_ parameters for configuring the logging, as described in the docs: https://esphome.io/components/logger.html
Logs are visible in real time from Home Assistant or by visiting the ESP32 with a web browser (if `web_server:` is in the configuration file). By default, logs are not saved anywhere.
I hope that helps!

**Comment by drkunze** [comment]
Thanks, I got that it enabled logging. How do you write the log file and where would it be?
On Jun 21, 2024 at 5:11 PM -0400, Nathan Borson ***@***.***>, wrote:

Hello @drkunze<https://github.com/drkunze>. If I understand your question right, nothing needs to go into it; just the one line is enough to enable logging with default settings.
logger:
There are various optional parameters for configuring the logging, as described in the docs: https://esphome.io/components/logger.html
Logs are visible in real time from Home Assistant or by visiting the ESP32 with a web browser (if web_server: is in the configuration file). By default, logs are not saved anywhere.
I hope that helps!

**Reply by nborson** [reply]
As far as I can tell, there is no log file. Rather, "The logger component automatically logs all log messages through the serial port and [only if MQTT is set up] through MQTT topics." There's just a feed, no persistent storage.
It sounds like there are ways to enable logging to a file, but it's more complicated than editing the yaml file:
https://community.home-assistant.io/t/esphome-logger-to-a-file/245946

**Comment by drkunze** [comment]
Thank you, I was hoping for an easy way to write to file, but I’m sure I don’t care enough for that:)

Thanks
On Jun 22, 2024 at 11:12 AM -0400, Nathan Borson ***@***.***>, wrote:

As far as I can tell, there is no log file. Rather, "The logger component automatically logs all log messages through the serial port and [only if MQTT is set up] through MQTT topics." There's just a feed, no persistent storage.
It sounds like there are ways to enable logging to a file, but it's more complicated than editing the yaml file:
https://community.home-assistant.io/t/esphome-logger-to-a-file/245946

**Comment by jeremyjhartley** [comment]
I was wanting to create a similar project, but it looks like you have done a majority of the work. I had a different thought for implementation however. I was wanting to keep the pump and the soil sensor separate. It is convenient that they are premade and sold as a unit. But I am just thinking that a submersible pump inside of the water container along with a water sensor to alert you when the container is empty might be more practical. I don't really want a pump sticking up out of the pot. Have you considered adding a sensor of some sort to alert you when the water container is empty?
